crates/app-data/src/app_data.rs

```diff
@@ -13,33 +11,40 @@ mod compat;
 /// The minimum valid empty app data JSON string.
 pub const EMPTY: &str = "{}";
 
-#[derive(Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct ValidatedAppData {
     pub hash: AppDataHash,
     pub document: String,
     pub protocol: ProtocolAppData,
 }
 
-#[derive(Debug, Default, Deserialize, Eq, PartialEq)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct ProtocolAppData {
     #[serde(default)]
     pub hooks: Hooks,
     pub signer: Option<H160>,
     pub replaced_order: Option<ReplacedOrder>,
+    pub partner_fee: Option<PartnerFee>,
 }
 
-#[derive(Debug, Default, Deserialize, Eq, PartialEq)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq)]
 pub struct ReplacedOrder {
     pub uid: OrderUid,
 }
 
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq)]
+pub struct PartnerFee {
```

Comment:
Are we sure this is the only fee we'll support? If not it might make sense to have this be an enum to avoid breaking changes in the future.

Reply:
@MartinquaXD good question! according to the [add-data documentation](https://github.com/cowprotocol/app-data/blob/main/src/schemas/partnerFee/v0.1.0.json) , this field is not an `enum`. If we have extra fees in the future, are they exclusive or we can have multiple of them?